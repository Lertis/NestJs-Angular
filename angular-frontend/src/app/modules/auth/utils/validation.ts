import { each } from 'lodash';
import { fieldValuesChanged } from '../../../shared/utils/common';

export function authFormValidation(
	current: { username: string; password: string },
	original: { username: string; password: string }
) {
	const valid: boolean[] = [
		!!current.username && current.username.trim().length !== 0 && current.password !== '',
		!!current.password && current.password.trim().length !== 0 && current.password !== '',
	];

	const changed: boolean[] = fieldValuesChanged(current, original);

	return {
		valid: valid.every((flag: boolean) => flag),
		changed: changed.some((flag: boolean) => !flag),
	};
}
