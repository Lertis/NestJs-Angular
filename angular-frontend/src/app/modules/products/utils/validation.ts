import { fieldValuesChanged } from '../../../shared/utils/common';

export function productProfileValidation(
	current: { description: string; price: number; title: string },
	original: { description: string; price: number; title: string }
) {
	const valid: boolean[] = [
		!!current.description && current.description.trim().length !== 0 && current.description !== '',
		!!current.title && current.title.trim().length !== 0 && current.title !== '',
		current.price !== 0,
	];

	const changed: boolean[] = fieldValuesChanged(current, original);

	return {
		valid: valid.every((flag: boolean) => flag),
		changed: changed.some((flag: boolean) => !flag),
	};
}
