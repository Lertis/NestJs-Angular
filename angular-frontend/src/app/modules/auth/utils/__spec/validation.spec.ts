import { AUTH_FORMS_COMPARE } from './validation.set';
import { authFormValidation } from '../validation';

declare const expect: jest.Expect;

describe('[Function]: authFormValidation', () => {
	AUTH_FORMS_COMPARE.forEach((test) => {
		it(test.testName, () => {
			const result = authFormValidation(test.current, test.original);
			expect(result).toMatchSnapshot();
		});
	});
});
