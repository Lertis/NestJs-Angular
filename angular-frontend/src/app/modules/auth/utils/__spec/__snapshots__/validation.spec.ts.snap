// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Function]: authFormValidation changed and invalid 1`] = `
Object {
  "changed": true,
  "valid": false,
}
`;

exports[`[Function]: authFormValidation changed and valid 1`] = `
Object {
  "changed": true,
  "valid": true,
}
`;

exports[`[Function]: authFormValidation doesn't changed and invalid 1`] = `
Object {
  "changed": false,
  "valid": false,
}
`;
